app/assets/scripts/components/table.js

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import T from 'prop-types';
+
+/*
+ * Table component
+ * @param data - data object used to construct table
+ *    expected in the following format:
+ *    {
+ *      header1: {
+ *        values: [Array](required),
+ *        formatCell: [function](optional),
+ *        formatHeader: [function])(optional),
+ *        sortable: [boolean](optional, default false),
+ *        cellWidth: [string](optional)
+ *      },
+ *      ...,
+ *      headerN: {
+ *        values: [Array](required),
+ *        formatCell: [function](optional),
+ *        formatHeader: [function])(optional),
+ *        sortable: [boolean](optional, default false),
+ *        cellWidth: [string](optional)
+ *      }
+ *    }
+ */
+const Table = styled.table``;
+const Row = styled.tr`
+  background-color: ${({ index }) => (index % 2 === 0 ? '#ffffff' : '#f4f9fe')};
+  ${({ header }) =>
+    header &&
+    css`
+      background-color: #f8f8f8;
+    `}
+  th:first-child {
+    border-left: none;
+  }
+  th:last-child {
+    border-right: none;
+  }
+`;
+const Header = styled.th`
+  max-width: 5rem;
+  padding: 0.75rem 1rem;
+  text-align: center;
+  border-left: 1px solid #e1e1e1;
+  border-right: 1px solid #e1e1e1;
+  border-collapse: collapse;
+  font-weight: 500;
+  font-size: 0.7rem;
+  ${({ active }) =>
+    active &&
+    css`
+      font-weight: 900;
+    `}
+
+  &:hover {
+    transform: scale(1.05);
+  }
+`;
+const Cell = styled.td`
+  padding: 0.75rem 1.5rem;
+
+  ${({ style }) =>
+    style &&
+    css`
+      ${Object.entries(style)
+        .map(([key, value]) => `${key}: ${value};`)
+        .join('\n')}
+    `}
+`;
+
+const prepareRows = data => {
+  const numVals = Math.max(
+    ...Object.values(data).map(datum => datum.values.length)
+  );
+  const rows = [];
+  for (let i = 0; i < numVals; i++) {
+    rows.push(
+      Object.entries(data).map(([key, column]) => {
+        return {
+          header: key,
+          value: column.values[i],
+          formatCell: column.formatCell,
+          style: column.style,
+        };
+      })
+    );
+  }
+  return rows;
+};
+function TableComponent(props) {
+  const { data } = props;
+  const [activeHeader, setActiveHeader] = useState(null);
+  const [sortAsc, setSortAsc] = useState(true);
+  return (
+    <Table className="global-table">
+      <thead>
+        <Row header>
+          {Object.entries(data).map(([key, datum]) => {
+            const { formatHeader } = datum;
+            return (
+              <Header
+                className={`sort_${sortAsc ? 'asc' : 'desc'}`}
+                key={key}
+                active={key === activeHeader}
+                onClick={() => {
+                  setActiveHeader(key);
+                  setSortAsc(!sortAsc);
```

Comment:
Looks like you'd actually want to split the state to control the columns sort individually?

Reply:
What do you think the functionality should be? I've implemented as follows but not sure it is most intuitive

click on a header -> toggles sorting direction and sets sorting to be based on that column

cc @tania-pires @AliceR